ares/ms/cartridge/board/hicom.cpp

```diff
@@ -0,0 +1,44 @@
+
+struct Hicom : Interface {
+  using Interface::Interface;
+  Memory::Readable<n8> rom;
+
+  auto load() -> void override {
+    Interface::load(rom, "program.rom");
+  }
+
+  auto save() -> void override {
+  }
+
+  auto unload() -> void override {
+  }
+
+  auto read(n16 address, n8 data) -> n8 override {
+    if(address <= 0x7fff)
+      return rom.read( (m_rom_bank_base * 0x8000) | (address&0x7fff));
+      
+    return data;
+  }
+
+  auto write(n16 address, n8 data) -> void override {
+	if (address == 0xffff){
+		m_rom_bank_base = data % (m_rom_page_count << 1);
+		print(m_rom_bank_base);
```

Comment:
please try to cleanup debug prints before submitting

Reply:
Oh i have no idea how i missed that one, will do all these as soon as i get back and finish one last mapper (just a small switching line needs to be worked out.)